CONTRIBUTING.md

```diff
@@ -0,0 +1,26 @@
+### Before you Start
+- Unless stated in README.md. Make sure you are pulling the source code from main branch with recursive variable to pull all submodules Collapse need to run.
+
+    ```git pull https://github.com/neon-nyan/Collapse --recursive```
+    
+- Make sure you use the supported IDE/SDKs listed below.
+- Make sure to read the Feature(s) restriction below.
+- When its time to do a Pull Request, do a pull request to the same branch when you first pulled.
+- We don't require a fully descriptive PR description, but please be clear on what added/removed and what it does.
+
+### Software Needed

```

Comment:
How about changing it to "Required Tools"? While it is software, I think the latter has a nicer ring to it and allows us to encompass a broader range of things in the future, should we need to.

Reply:
That's actually a good idea, applied changes and adjusted other sections to match this one